trusted.ci.jenkins.io.tf

```diff
@@ -122,6 +124,74 @@ resource "azurerm_linux_virtual_machine" "trusted_bounce" {
   }
 }
 
+# CONTROLLER VM
+## NETWORK INTERFACE with internal ip
+resource "azurerm_network_interface" "trusted_controller" {
+  name                = "trusted-controller"
```

Comment:
Not blocking this PR, but need to be addressed before migration to this new machine:

The naming `trusted-*` for these VMs can be confusing:

- Is `trusted` an adjective or a name?
- How should we name `ci.jenkins.io` VM? `ci-controller`? Then it's not coherent with the naming here.

Proposal: ensure that the `ci` string is included, but do not add the `jenkins.io` domain as it is subject to change (and add too much elements in the name). WDYT?

- `trusted-ci-controller` (Terraform resource name: `trusted_ci_controller`)
- For ci.jenkins.io: `ci-controller` (`ci_controller`)
- For cert.ci.jenkins.io): `cert-ci-controller` (`trusted_ci_jenkins_io_controller`)
- Bounce would be named to `trusted-ci-bounce` (I would even argue for `trusted-ci-bastion`)

Reply:
https://github.com/jenkins-infra/azure/pull/331#pullrequestreview-1398165722 would make this a lot simpler and less to review imo

Reply:
@timja I'm having mixed feelings about a module as the 3 controllers have 3 very different infrasturcture topologies: 1 public, 1 private with SSH bounce and 1 private behind a VPN.

Not sure about the value of "DRY" as most of the lines have different values :|

Reply:
(btw sorry we missed your comment on the closed PR: do you mind commenting on the issue to have the discussion in a place were it would be easier to keep and track it?)

Reply:
(Don't get me wrong, I don't see a module would not be usefull, I'm questionning the value "now" for the trusted migration: I would have that discussion with https://github.com/jenkins-infra/helpdesk/issues/3535 for instance, comparing if it is worth it with the trusted machine)